opal/mca/btl/tcp/btl_tcp_component.c

```diff
@@ -318,7 +318,9 @@ static int mca_btl_tcp_component_register(void)
     mca_btl_tcp_module.super.btl_rndv_eager_limit = 64*1024;
     mca_btl_tcp_module.super.btl_max_send_size = 128*1024;
     mca_btl_tcp_module.super.btl_rdma_pipeline_send_length = 128*1024;
-    mca_btl_tcp_module.super.btl_rdma_pipeline_frag_size = INT_MAX;
+    /* Some OSes have hard coded limits on how many bytes can be manipulated by each writev operation.
+     * Force a reasonable limit, to prevent overflowing a 32-bit integer (limit comes from BSD and OS X) */
+    mca_btl_tcp_module.super.btl_rdma_pipeline_frag_size = ((1UL<<31) - 1024);
```

Comment:
This really needs to be a #define and also, I'm not sure I can figure out exactly why the - 1024 from the comment; a bit more context as to why that instead of INT_MAX would be super useful.

Reply:
(1UL<<31) was found by empirically testing on different systems and then taking the lowest value (OS X). The 1024 was to have some slack before the strict limit imposed by some OSes. This number can certainly be lowered to the largest PML header, but in this particular instance I don't think such accuracy is necessary.

Reply:
Can you add that to the comment?  Otherwise, the whole series looks good to me.

Reply:
Comment has been updated to reflect the randomness of choice.